blobs/p8z77-m_pro/download_BIOS_clean.sh

```diff
@@ -0,0 +1,109 @@
+FINAL_GBE_SHA256SUM="fca4deb13633712113e1824bfd5afa32f487ca7129ca012fecf5d7502ec1d5ba  flashregion_3_gbe.bin"
+ZIPURL="https://dlcdnets.asus.com/pub/ASUS/mb/LGA1155/P8Z77-V_PRO/P8Z77-V-PRO-ASUS-2104.zip"
+
+ZIPFILENAME=`echo $ZIPURL | sed 's/.*\///'`
+ROMFILENAME=`echo $ZIPFILENAME | sed 's/\.zip$/\.ROM/'`
+
+extractdir=$(mktemp -d)
+echo "### Creating temp dir $extractdir "
+cd "$extractdir"
+
+/bin/cat <<EOF > layout.txt
+00000000:00000fff fd
+0001b000:007fffff bios
+00003000:0001afff me
+00001000:00002fff gb
+EOF
+
+
+echo "### Downloading $ZIPURL"
+wget $ZIPURL || { echo "ERROR: wget failed $ZIPURL" && exit 1; }
+echo "### Verifying expected hash of $ZIPFILENAME"
+echo "$CAP_ZIP_SHA256SUM" | sha256sum --check || { echo "Failed sha256sum verification on downloaded binary..." && exit 1; }
+
+echo "### Extracting Archives"
+unzip $ZIPFILENAME || { echo "Failed unzipping $ZIPFILENAME - Tool installed on host?" && exit 1;}
+
+echo "### Verifying expected hash of $ROMFILENAME"
+echo "$CAP_FILE_SHA256SUM" | sha256sum --check || { echo "Failed sha256sum verification on extracted binary..." && exit 1; }
+
+echo "### extracing BIOS from Capsule"
+dd bs=1024 skip=2 if=P8Z77-V-PRO-ASUS-2104.CAP of=P8Z77-V-PRO-ASUS-2104.ROM || { echo "Failed to de-cap the ROM..." && exit 1; }
+
+echo "### Stock variant AltME & ME Cleaner"
+$IFDTOOL -M 1 $ROMFILENAME
+echo "### extract stock ME"
+$IFDTOOL -x $ROMFILENAME.new
+
+echo "### Applying me_cleaner to neuter and truncate. EFFS,FCRS whitelisted"
+$MECLEAN -r -t -O "$BLOBDIR/me.bin" flashregion_2_intel_me.bin
+
+echo "### Verifying expected hash of me.bin"
+echo "$FINAL_ME_SHA256SUM" | sha256sum --check || { echo "Failed sha256sum verification on final binary..." && exit 1; }
+rm flashregion*.bin
+echo "### Resizing ..."
+$IFDTOOL -D 8 $ROMFILENAME.new || { echo "Failed density resize " && exit 1;}
+$IFDTOOL --newlayout layout.txt $ROMFILENAME.new.new || { echo "Failed new layout ..." && exit 1;}
+echo "### Extracting final IFD"
+$IFDTOOL -x $ROMFILENAME.new.new.new || { echo "Failed ifdtool. Tool installed on host?" && exit 1;}
+
+printf '\x00' | dd of=flashregion_0_flashdescriptor.bin bs=1 seek=3837 count=1 conv=notrunc
+printf '\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF' | dd of=flashregion_0_flashdescriptor.bin bs=1 seek=3568 count=32 conv=notrunc
+
+echo "### Verifying expected hash of IFD"
+echo "$FINAL_IFD_SHA256SUM" | sha256sum --check || { echo "Failed sha256sum verification on factory IFD bin..." && exit 1; }
+cp flashregion_0_flashdescriptor.bin $BLOBDIR/ifd.bin || { echo "Failed to copy IFD ..." && exit 1; }
+echo "$FINAL_GBE_SHA256SUM" | sha256sum --check || { echo "Failed sha256sum verification on factory IFD bin..." && exit 1; }
+cp flashregion_3_gbe.bin $BLOBDIR/gbe.bin || { echo "Failed to copy GBE ..." && exit 1; }
```

Comment:
Confused here. What is the hardcoded mac address? Any? From my understanding, those needed to be generated otherwise coming from user's taken backups of their own ROM.

I was not aware that the manufacturer would/could provide in their UEFI capsule an IFD and even less a GBE config: which normally includes a MAC address. This is interesting, to say the least.

Reply:
ifdtool dependency removed, script updated.

Reply:
just took a look for you, I dont see a pre defined MAC in gbe.bin, 0x1000 onwards are 0xFF. Looking at your generated xx30 gbe.bin its clear that what I extract is definitely a gbe.bin file (ex 0xAD 0xBA from 0x40 etc etc)

I don't use networking in coreboot personally and having the GBE partition in this state does not appear to affect operation when booing an linux based OS

Reply:
hold on.. think this may be on me for GBE.bin ... ive had a usb network adapter connected to my test machine. #fail. 

the onboard nic isnt showing up. 

please hold this PR for now till i figure out a solution.

Reply:
see https://github.com/osresearch/heads/pull/1345#issuecomment-1476777824 

seems fine